react/features/facial-recognition/actions.js

```diff
@@ -52,80 +57,99 @@ let duplicateConsecutiveExpressions = 0;
-            // it is received after as a response to the first message sent to the worker.
-            if (type === INTERVAL_MESSAGE) {
-                value && dispatch(setDetectionTimeInterval(value));
-            }
-
-            // receives a message with the predicted facial expression.
-            if (type === FACIAL_EXPRESSION_MESSAGE) {
-                sendDataToWorker(worker, imageCapture);
-                if (!value) {
-                    return;
-                }
-                if (value === lastFacialExpression) {
+            if (faceExpression) {
+                if (faceExpression === lastFacialExpression) {
                     duplicateConsecutiveExpressions++;
                 } else {
                     if (lastFacialExpression && lastFacialExpressionTimestamp) {
-                        dispatch(
-                        addFacialExpression(
+                        dispatch(addFacialExpression(
                             lastFacialExpression,
                             duplicateConsecutiveExpressions + 1,
                             lastFacialExpressionTimestamp
-                        )
-                        );
+                        ));
                     }
-                    lastFacialExpression = value;
+                    lastFacialExpression = faceExpression;
                     lastFacialExpressionTimestamp = Date.now();
                     duplicateConsecutiveExpressions = 0;
                 }
             }
+
+            if (faceBox) {
+                const state = getState();
+                const conference = getCurrentConference(state);
+                const localParticipant = getLocalParticipant(state);
+
+                if (getParticipantCount(state) > 1) {
+                    sendFaceBoxToParticipants(conference, faceBox);
+                }
+
+                dispatch({
+                    type: UPDATE_FACE_COORDINATES,
+                    faceBox,
+                    id: localParticipant.id
+                });
+            }
         };
+
+        const { enableFacialRecognition, faceCoordinatesSharing } = getState()['features/base/config'];
+        const detectionTypes = [
+            faceCoordinatesSharing?.enabled && DETECTION_TYPES.FACE_BOX,
+            enableFacialRecognition && DETECTION_TYPES.FACE_EXPRESSIONS
+        ].filter(Boolean);
+
         worker.postMessage({
             type: INIT_WORKER,
-            url: baseUrl,
-            windowScreenSize: window.screen ? {
-                width: window.screen.width,
-                height: window.screen.height
-            } : undefined
+            baseUrl,
+            detectionTypes
         });
+
         dispatch(startFacialRecognition());
     };
 }
 
 /**
  * Starts the recognition and detection of face expressions.
  *
- * @param  {Object} stream - Video stream.
+ * @param {Track | undefined} track - Track for which to start detecting faces.
```

Comment:
When would the track de defined and when not?

Reply:
we only pass the track when we intercept it changes in the middleware, at which point it wouldn't be available yet in the state